Simulator/Recorders/Hdf5Recorder.cpp

```diff
@@ -122,6 +123,94 @@ void Hdf5Recorder::saveSimData(const AllVertices &neurons)
+               variableInfo.captureData();
+            } else {
+               // Handle the case where the dataset already exists
+
+               // Get the current dimensions of the dataset
+               DataSpace currentSpace = variableInfo.hdf5DataSet_.getSpace();
+               hsize_t currentDims[1];
+               currentSpace.getSimpleExtentDims(currentDims, nullptr);
+
+               // Calculate the new dimensions after appending the new data
+               hsize_t appendDims[1] = {currentDims[0] + newDims[0]};
+               variableInfo.hdf5DataSet_.extend(appendDims);
+
+               // Select the hyperslab in the extended portion of the dataset
+               DataSpace fileSpace = variableInfo.hdf5DataSet_.getSpace();
+               hsize_t offset[1] = {currentDims[0]};
+               fileSpace.selectHyperslab(H5S_SELECT_SET, newDims, offset);
+
+               // Create a memory dataspace for the new data
+               DataSpace memSpace(1, newDims);
+
+               // Prepare the data buffer and write the new data to the dataset
+               if (variableInfo.hdf5Datatype_ == PredType::NATIVE_FLOAT) {
+                  std::vector<float> dataBuffer(variableInfo.variableLocation_.getNumElements());
+                  for (size_t i = 0; i < variableInfo.variableLocation_.getNumElements(); ++i) {
+                     dataBuffer[i] = get<float>(variableInfo.variableLocation_.getElement(i));
+                  }
+                  variableInfo.hdf5DataSet_.write(dataBuffer.data(), variableInfo.hdf5Datatype_,
+                                                  memSpace, fileSpace);
+               } else if (variableInfo.hdf5Datatype_ == PredType::NATIVE_INT) {
+                  std::vector<int> dataBuffer(variableInfo.variableLocation_.getNumElements());
+                  for (size_t i = 0; i < variableInfo.variableLocation_.getNumElements(); ++i) {
+                     dataBuffer[i] = get<int>(variableInfo.variableLocation_.getElement(i));
+                  }
+                  variableInfo.hdf5DataSet_.write(dataBuffer.data(), variableInfo.hdf5Datatype_,
+                                                  memSpace, fileSpace);
+               } else if (variableInfo.hdf5Datatype_ == PredType::NATIVE_UINT64) {
+                  std::vector<uint64_t> dataBuffer(variableInfo.variableLocation_.getNumElements());
+                  for (size_t i = 0; i < variableInfo.variableLocation_.getNumElements(); ++i) {
+                     dataBuffer[i] = get<uint64_t>(variableInfo.variableLocation_.getElement(i));
+                  }
+                  variableInfo.hdf5DataSet_.write(dataBuffer.data(), variableInfo.hdf5Datatype_,
+                                                  memSpace, fileSpace);
+               } else {
+                  // Throw an exception if the data type is unsupported
+                  throw std::runtime_error("Unsupported data type");
```

Comment:
Can you put a bit more information in the message? At least include the variable name.